amethyst_tiles/src/pass.rs

```diff
@@ -0,0 +1,444 @@
+
+        {
+            #[cfg(feature = "profiler")]
+            profile_scope!("write");
+            self.vertex.write(
+                factory,
+                index,
+                self.sprites.count() as u64,
+                Some(self.sprites.data()),
+            );
+
+            self.env.write(factory, index, tilemap_args.std140());
+        }
+
+        self.change.prepare_result(index, changed)
+    }
+
+    fn draw_inline(
+        &mut self,
+        mut encoder: RenderPassEncoder<'_, B>,
+        index: usize,
+        _subpass: hal::pass::Subpass<'_, B>,
+        _world: &World,
+    ) {
+        #[cfg(feature = "profiler")]
+        profile_scope!("draw");
+
+        let layout = &self.pipeline_layout;
+        encoder.bind_graphics_pipeline(&self.pipeline);
+        self.env.bind(index, layout, 0, &mut encoder);
+
+        self.vertex.bind(index, 0, 0, &mut encoder);
+        for (&tex, range) in self.sprites.iter() {
+            if self.textures.loaded(tex) {
+                self.textures.bind(layout, 1, tex, &mut encoder);
+
+                unsafe {
+                    encoder.draw(0..4, range);
+                }
+            }
+        }
+    }
+
+    fn dispose(self: Box<Self>, factory: &mut Factory<B>, _aux: &World) {
+        unsafe {
+            factory.device().destroy_graphics_pipeline(self.pipeline);
+            factory
+                .device()
+                .destroy_pipeline_layout(self.pipeline_layout);
+        }
+    }
+}
+
+fn build_tiles_pipeline<B: Backend>(
+    factory: &Factory<B>,
+    subpass: hal::pass::Subpass<'_, B>,
+    framebuffer_width: u32,
+    framebuffer_height: u32,
+    layouts: Vec<&B::DescriptorSetLayout>,
+) -> Result<(B::GraphicsPipeline, B::PipelineLayout), failure::Error> {
+    let pipeline_layout = unsafe {
+        factory
+            .device()
+            .create_pipeline_layout(layouts, None as Option<(_, _)>)
+    }?;
+
+    //let shader_vertex = unsafe { VERTEX.module(factory).unwrap() };
+    //let shader_fragment = unsafe { FRAGMENT.module(factory).unwrap() };
+    let mut shaders = SHADERS.build(factory, Default::default())?;
+
+    let pipes = PipelinesBuilder::new()
+        .with_pipeline(
+            PipelineDescBuilder::new()
+                .with_vertex_desc(&[(TileArgs::vertex(), pso::VertexInputRate::Instance(1))])
+                .with_input_assembler(pso::InputAssemblerDesc::new(hal::Primitive::TriangleStrip))
+                .with_shaders(shaders.raw()?)
+                .with_layout(&pipeline_layout)
+                .with_subpass(subpass)
+                .with_framebuffer_size(framebuffer_width, framebuffer_height)
+                .with_blend_targets(vec![pso::ColorBlendDesc(
+                    pso::ColorMask::ALL,
+                    pso::BlendState::PREMULTIPLIED_ALPHA,
+                )])
+                .with_depth_test(pso::DepthTest::On {
+                    fun: pso::Comparison::Less,
```

Comment:
I'm not sure if you want to write depth buffer when rendering sprites, transparency might get in the way. See the comment on region iterator in rendering pass.

Reply:
The tile map specifically supports transparent alpha channel tiles. I have been using it without issue, except for the drawing artifacts discussed previously. It seems to work fine, lets discuss offline more.

Reply:
Done